src/pocketmine/Player.php

```diff
@@ -593,6 +604,7 @@ public function setOp(bool $value){
 		}else{
 			$this->server->removeOp($this->getName());
 		}
+		$this->setRemoveFormat(!$value);
```

Comment:
This can be overridden by plugins, which is why I didn't apply such a fix myself already. Additionally this is not related to mute.